saleor/extensions/plugins/vatlayer/plugin.py

```diff
@@ -72,12 +74,9 @@ def calculate_checkout_total(
         if self._skip_plugin(previous_value):
             return previous_value
 
-        zero_total = Money(0, currency=previous_value.currency)
-        taxed_zero = TaxedMoney(zero_total, zero_total)
-
         return (
-            self.calculate_checkout_subtotal(checkout, discounts, previous_value)
-            + self.calculate_checkout_shipping(checkout, discounts, taxed_zero)
+            calculations.checkout_subtotal(checkout, discounts)
```

Comment:
Also - I am not sure of this. @krzysztofwolski Can you help?

Reply:
AFAIK `checkout .calculation` module is the `top` level of calculation.  So I am not sure that we should trigger it inside a plugin, which should respect hierachy `calculation` -> `manager` -> `plugin`

Reply:
I consulted this with @krzysztofwolski 
For what I understand we want to run `checkout_subtotal` for every plugin
and this should do it.
But we can discuss this if we want something different, or if we should use different approach

Reply:
Maybe I am wrong but let's take as an example `checkout_shipping_price`.
All Saleor logic uses `calculation.checkout_shipping_price`, this calls manager and manager calls each plugin one by one.
The same flow is for the rest methods. Also, Avatax uses the same flow.

Reply:
In your case we have: `calculation.checkout_subtotal` -> `manager` -> plugin one by one -> `Vatlayer` -> `calculation.checkout_subtotal`

Reply:
Ok, my bad.